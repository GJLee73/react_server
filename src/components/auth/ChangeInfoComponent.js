import React, { Component } from 'react';
import {autoHypenPhone,verifyEmail} from '../../function/registers'
class ChangeInfoComponent extends Component {
    constructor(props) {
        super(props);
        this.state = { value:this.props.value}
        this.handleSend=this.handleSend.bind(this)
        this.handleChange=this.handleChange.bind(this)
    }
    handleChange(value){
        this.setState({value:value})
    }
    handleSend(){
        if(this.props.typeName=='email'){
            if(!verifyEmail(this.state.value))
            return 0
        }else if(this.props.typeName=='phone'){
            if(this.state.value.length<12){
				alert('정확한 길이의 번호를 입력하세요')
				return 0;
			}
        }
        var msg ={}
        msg[this.props.typeName]=this.state.value
        this.props.submit(msg)
    }
       
    render() { 
        const checkType=()=>{
            switch(this.props.typeName){
                case 'phone' : 
                return  <input onChange={(e)=>{
                    var phone = autoHypenPhone(e.target.value)
                    this.handleChange(phone)
                }}
                     value={this.state.value} maxLength={13}/>
                     default:
                     return  <input onChange={(e)=>this.handleChange(e.target.value)} value={this.state.value}/> 
            }
        }
        return ( <div>
           {checkType()}
            <button onClick={this.handleSend}>완료</button> </div> )
    }
}
 
export default ChangeInfoComponent;