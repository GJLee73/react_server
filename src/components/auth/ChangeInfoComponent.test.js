import React from 'react';
import {shallow} from 'enzyme'
import ChangeInfoComponent from './ChangeInfoComponent'

describe('snapshot, event test',()=>{
    it('snapshot test',()=>{
       
        var wrapper =shallow(<ChangeInfoComponent value ='g1' typeName='nickName'/>)
       
        expect(wrapper).toMatchSnapshot()
    })
    it('event test',()=>{
        var handleChange = jest.spyOn(ChangeInfoComponent.prototype,'handleChange')
        var handleSend = jest.spyOn(ChangeInfoComponent.prototype,'handleSend')
        var submit = jest.fn()
        var wrapper =shallow(<ChangeInfoComponent value ='g1' typeName='nickName' submit={submit}/>)
        var mockedEvent = {target:{value:'g2'}}
        wrapper.find('input').simulate('Change',mockedEvent)
        console.log(wrapper.find('input'))
        expect(handleChange.mock.calls.length).toBe(1)
        expect(wrapper.state('value')).toBe('g2')
        wrapper.find('button').simulate('click')
        expect(handleSend.mock.calls.length).toBe(1)
        expect(submit.mock.calls.length).toBe(1)
    })
   
})