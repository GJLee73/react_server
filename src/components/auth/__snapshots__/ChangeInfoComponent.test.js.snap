// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot, event test snapshot test 1`] = `
<div>
  <input
    onChange={[Function]}
    value="g1"
  />
  <button
    onClick={[Function]}
  >
    완료
  </button>
   
</div>
`;
